internal/graph/check.go

```diff
@@ -18,6 +17,64 @@ import (
 
 var tracer = otel.Tracer("internal/graph/check")
 
+// CheckResolver represents an interface that can be implemented to provide recursive resolution
+// of a Check.
+type CheckResolver interface {
```

Comment:
I'll just leave this here https://go.dev/doc/effective_go#interface-names, and say how much I would like a `Checker` interface!

Reply:
Talked offline. The reason for not naming this interface `Checker` is because then that will lead to obscurities with our existing `Check` interface for the gRPC server definition. We'll have

`Check(ctx context.Context, req *openfgapb.CheckRequest) (*openfgapb.CheckResponse, error)`
and
`Check(ctx context.Context, req *graph.CheckRequest) (*graph.CheckResponse, error)`

I purposefully chose `CheckResolver` isntead of simply `Checker` because it avoids these overlapping signatures. The only distinguishing piece above is the package prefix (`openfgapb` vs `graph`). With this interface name we avoid the signature collisions altogether.

`ResolveCheck(ctx context.Context, req *graph.ResolveCheckRequest) (*graph.ResolveCheckResponse, error)`.